source/game.c

```diff
@@ -2203,287 +2207,349 @@ static bool check_and_score_joker_for_event(ListItr* starting_joker_itr, CardObj
-                                timer = TM_ZERO;
-                                break; // Break out of the loop to avoid accessing an invalid index
-                            }
-                        }
-
-                        *discarded_card = true;
-                    }
+static void played_cards_update_loop()
+{
+    // So this one is a bit fucking weird because I have to work kinda backwards for everything because of the order of the pushed cards from the hand to the play stack
+    // (also crazy that the company that published Balatro is called "Playstack" and this is a play stack, but I digress)
+    for (int played_idx = 0; played_idx <= played_top; played_idx++)
+    {
+        if (played[played_idx] == NULL)
+        {
+            continue;
+        }
 
-                    break;
-            }
+        if (card_object_get_sprite(played[played_idx]) == NULL)
+        {
+            card_object_set_sprite(played[played_idx], played_idx + MAX_HAND_SIZE); // Set the sprite for the played card object
+        }
+
+        switch (play_state)
+        {
+            case PLAY_STARTING:
+
+                play_starting_played_cards_update(played_idx);
+                break;
+            
+            case PLAY_BEFORE_SCORING:
+
+                if (play_before_scoring_cards_update(played_idx))
+                {
+                    return;
+                }
+                break;
+
+            case PLAY_SCORING_CARDS:
+
+                if (play_scoring_cards_update())
+                {
+                    return;
+                }
+                break;
+            
+            case PLAY_SCORING_CARD_JOKERS:
+
+                if (play_scoring_card_jokers_update())
+                {
+                    return;
+                }
+                break;
+            
+            case PLAY_SCORING_HELD_CARDS:
 
-            played[i]->sprite_object->tx = played_x;
-            played[i]->sprite_object->ty = played_y;
-            played[i]->sprite_object->tscale = played_scale;
-            card_object_update(played[i]);
+                if (play_scoring_held_cards_update(played_idx))
+                {
+                    return;
+                }
+                break;
+
+            case PLAY_SCORING_INDEPENDENT_JOKERS:
+
+                if (play_scoring_independent_jokers_update(played_idx))
+                {
+                    return;
+                }
+                break;
+
+            case PLAY_ENDING:
+
+                play_ending_played_cards_update(played_idx);
+                break;
+
+            case PLAY_ENDED:
+
+                if (play_ended_played_cards_update(played_idx))
+                {
+                    // we continue here instead of returning to instantly go to
+                    // the next card to discard, instead of starting over
+                    // from index 0 and going up to that card again
+                    continue;
```

Comment:
```suggestion
                    // we continue here instead of returning for performance 
                    // to instantly go to the next card to discard at played_idx+1, 
                    // instead of  starting over from index 0 and going up
                    // to that card again 
                    continue;
```

Updated the comment a bit to clarify that it's for performance, and added explicit `played_idx+1` from your explanation just in case it makes it clearer.

Reply:
No no that's precisely what I meant to do there ^^